src/WebJobs.Extensions.DurableTask/DurableTaskExtension.cs

```diff
@@ -109,7 +118,14 @@ public DurableTaskExtension()
             IDurableHttpMessageHandlerFactory durableHttpMessageHandlerFactory = null,
             ILifeCycleNotificationHelper lifeCycleNotificationHelper = null,
             IMessageSerializerSettingsFactory messageSerializerSettingsFactory = null,
+#if !FUNCTIONS_V1
+            IErrorSerializerSettingsFactory errorSerializerSettingsFactory = null,
+#pragma warning disable SA1113, SA1001, SA1115
+            ITelemetryActivator telemetryActivator = null)

```

Comment:
From what I can tell this is never _not_ provided, at least in production, because we inject it via Dependency Injection. If it is null in our test code, let's just feed in a no-op version of the interface.

Reply:
By "this" did you mean the `telemetryActivator` or the `errorSerializerSettingsFactory `?

Reply:
I've also been playing with how to simplify this conditional compilation bit. It looks like it could be simplified since `errorSerializerSettingsFactory` is always present, but there's no clean way to add it without violating linter-rules. There are also a few places where this constructor's optional parameters are called without explicitly specifying the parameter's names, so the order of params needs to remain intact.

I would personally vote to keep this _as-is_, for now.

Reply:
I was referring to `telemetryActivator`, but I think I am fine leaving it for now.